tests/test_gaussian.py

```diff
@@ -73,10 +73,13 @@ def SF_gate_reference(sf_op, wires, *args):
 def SF_expectation_reference(sf_expectation, wires, *args):
     """SF reference circuit for expectation tests"""
     eng = sf.Engine("gaussian")
-    prog = sf.Program(2)
+
+    # Allows returning the variance of tensor number for 3 modes
+    num_wires = 2 if len(wires) <=2 else 3
```

Comment:
This line seems a bit unclear 🤔 Should the number of wires always correspond to `len(wires)`?

Reply:
Valid point! Certain tests have `wires=[0]` (e.g. `test_one_mode_single_real_parameter_gates`) and use `SF_gate_reference`. For those this would be failing. Would it be better to have two separate `SF_gate_reference` definitions?

Reply:
Ah, I see. I think it would be better to simply have `num_wires = len(wires)` here, and then below we can do

```python
with prog.context as q:
    sf.ops.Dgate(0.1) | q[0]
    if num_wires > 1:
            sf.ops.S2gate(0.1) | (q[0], q[1])
```

Everything should continue to work, but the logic is more readable

Reply:
One thing to note is that we'd like to create `Program(2)` even if `wires=[0]` (`wires` is used for `sf_expectation(state, wires, args)`).

Added a new `num_wires` parameter to the signature of `SF_expectation_reference`.